packages/discovery-react-components/src/components/DocumentPreview/components/HtmlView/HtmlView.tsx

```diff
@@ -53,13 +55,46 @@ export const HtmlView: FC<Props> = ({
   }
 
   const [html, setHtml] = useState<string | null>(null);
+  const [processedDoc, setProcessedDoc] = useState<ProcessedDoc | null>(null);
+  const [locationArray, setLocationArray] = useState<Array<Location> | null>(null);
+
+  const textMappingBbox = usePassage(document, highlight as QueryResultPassage);
+  useEffect(() => {
+    if (highlight) {
+      if (isPassage(highlight)) {
+        if (processedDoc && textMappingBbox) {
+          let processedDocBbox: ProcessedBbox[] = [];
+          textMappingBbox &&
```

Comment:
You're already checking if `textMappingBbox` is truthy in the `if` above, so this isn't necessary.

Also, these compound `&&` statements make the code difficult to read. Much better to write it out using `if` blocks.